src/MultiValueControl.tsx

```diff
@@ -0,0 +1,173 @@
+import { Checkbox } from "office-ui-fabric-react/lib/components/Checkbox";
+import { ITag, TagPicker } from "office-ui-fabric-react/lib/components/pickers";
+import { TextField } from "office-ui-fabric-react/lib/components/TextField";
+import { FocusZone, FocusZoneDirection } from "office-ui-fabric-react/lib/FocusZone";
+import * as React from "react";
+import { DelayedFunction } from "VSS/Utils/Core";
+import { BrowserCheckUtils } from "VSS/Utils/UI";
+
+interface IMultiValueControlProps {
+    selected?: string[];
+    width?: number;
+    readOnly?: boolean;
+    placeholder?: string;
+    noResultsFoundText?: string;
+    searchingText?: string;
+    onSelectionChanged?: (selection: string[]) => Promise<void>;
+    forceValue?: boolean;
+    options: string[];
+    onBlurred?: () => void;
+    onResize?: () => void;
+}
+
+interface IMultiValueControlState {
+    focused: boolean;
+    filter: string;
+}
+
+export class MultiValueControl extends React.Component<IMultiValueControlProps, IMultiValueControlState> {
+    private readonly _unfocusedTimeout = BrowserCheckUtils.isSafari() ? 2000 : 1;
+    private _setUnfocused = new DelayedFunction(null, this._unfocusedTimeout, "", () => {
+        this.setState({focused: false, filter: ""});
+    });
+    constructor(props, context) {
+        super(props, context);
+        this.state = { focused: false, filter: "" };
+    }
+    public render() {
+        const {focused} = this.state;
+        return <div className={`multi-value-control ${focused ? "focused" : ""}`}>
+            <TagPicker
+                className="tag-picker"
+                selectedItems={(this.props.selected || []).map((t) => ({ key: t, name: t }))}
+                inputProps={{
+                    placeholder: this.props.placeholder,
+                    readOnly: this.props.readOnly,
+                    width: this.props.width || 200,
+                    onFocus: () => this.setState({ focused: true }),
+                }}
+                onChange={this._onTagsChanged}
+                onResolveSuggestions={() => []}
+            />
+            {focused ? this._getOptions() : null}
+        </div>;
+    }
+    public componentDidUpdate() {
+        if (this.props.onResize) {
+            this.props.onResize();
+        }
+    }
+    private _getOptions() {
+        const options = this.props.options;
+        const selected = (this.props.selected || []).slice(0);
+        const filteredOpts = this._filteredOptions();
+
+        return <div className="options">
+            <TextField value={this.state.filter}
```

Comment:
Cant you just use vss-ui picklist for this?

![image](https://user-images.githubusercontent.com/3766956/44417243-d9231f80-a529-11e8-9711-7f37754c6bd2.png)

Reply:
![image](https://user-images.githubusercontent.com/3766956/44417273-eb9d5900-a529-11e8-85af-70adb9f77a26.png)

Reply:
Dropdown's in general are buggy on form group extensions because of their proclivity to get clipped by the iframes borders. The picklist from vss-ui in particular lacks the ability to
- fire an event when the search is performed so that the iframe can be resized
- respond to keydown when only one element matches search - currently used to toggle selection.
- reorder search results so that the elements that start with the search query show up first.